chirho/counterfactual/handlers/explanation.py

```diff
@@ -0,0 +1,131 @@
+import contextlib
+from typing import Callable, Iterable, Mapping, Optional, ParamSpec, TypeVar
+
+import pyro
+import pyro.distributions
+import torch
+
+from chirho.counterfactual.handlers.counterfactual import BiasedPreemptions
+from chirho.counterfactual.handlers.selection import get_factual_indices
+from chirho.indexed.ops import IndexSet, cond, gather, indices_of, scatter
+from chirho.interventional.handlers import do
+from chirho.interventional.ops import Intervention
+from chirho.observational.handlers import condition
+from chirho.observational.handlers.condition import Factors
+from chirho.observational.ops import Observation
+
+P = ParamSpec("P")
+T = TypeVar("T")
+
+
+def factual_preemption(
```

Comment:
The helper functions `factual_preemption` and `consequent_differs_factor` seem uncontroversial and can probably be tested and merged into `master` once #238 and #239 land.